common/models/role.js

```diff
@@ -66,8 +77,14 @@ module.exports = function(Role) {
           roles: ACL.ROLE,
         };
 
-        var model = relsToModels[rel];
-        listByPrincipalType(this, model, relsToTypes[rel], query, callback);
+        // resolve user model and user type if principalType is available and not "USER"
+        if (rel === 'users' &&
+          query.where.principalType && query.where.principalType !== RoleMapping.USER) {
+          relsToModels.users = this.constructor.registry.getModel(query.where.principalType);
+          relsToTypes.users = query.where.principalType;
```

Comment:
i wanted to stick to existing code style by not modifying the following method call in
`listByPrincipalType(this, relsToModels[rel], relsToTypes[rel], query, callback);`

but your proposal perfectly suits my mood